source/time_tables/departure_boards.cpp

```diff
@@ -92,6 +92,51 @@ render(PbCreator& pb_creator,
     }    
 }
 
+time_duration to_navitia (boost::posix_time::time_duration dur) {
+    return navitia::seconds(dur.total_seconds());
+}
+
+time_duration length_of_time(const time_duration& duration_1,
+                       const time_duration& duration_2) {
+    if (duration_1 <= duration_2) {
+        return duration_2 - duration_1;
+    } else {
+        return duration_2 + (hours(24) - duration_1);
+    }
+}
+
+bool between_opening_and_closing(const time_duration& me, const time_duration& opening, const time_duration& closing) {
+    if (opening < closing) {
+        return (opening <= me && me <= closing);
+    } else {
+        return (opening <= me || me <= closing);
+    }
+}
+
+bool line_closed (const time_duration& duration,
+                  const time_duration opening,
+                  const time_duration closing,
+                  const pt::ptime& date ) {
+    const auto begin = to_navitia(date.time_of_day());
+    if (!between_opening_and_closing(begin, opening, closing)
+            && !between_opening_and_closing((begin + duration), opening, closing)) {
+        if (duration < length_of_time(opening, closing) + length_of_time(begin, opening)) {
+            return true;
+        }
```

Comment:
you can do `return duration < length_of_time(opening, closing) + length_of_time(begin, opening)`